pkg/virt-controller/services/template.go

```diff
@@ -976,12 +983,23 @@ func getNetworkToResourceMap(virtClient kubecli.KubevirtClient, vmi *v1.VirtualM
 
 func getCniInterfaceList(vmi *v1.VirtualMachineInstance) (ifaceListString string, cniAnnotation string) {
 	ifaceList := make([]string, 0)
+	ifaceListMap := make([]map[string]string, 0)
 
-	for _, network := range vmi.Spec.Networks {
+	for idx, network := range vmi.Spec.Networks {
 		// set the type for the first network
 		// all other networks must have same type
 		if network.Multus != nil {
-			ifaceList = append(ifaceList, network.Multus.NetworkName)
+			networkSource := v1.NetworkSource{Npwgv1: network.Multus}
```

Comment:
It seems to be a pattern in this PR to convert network spec in-place over and over. I don't think we really need to do this. You can introduce a helper function "getNpwgNetwork" that, when called on ```multus``` network, would return a structure of ```npwg``` type, and would return as-is for an ```npwg``` networks. There seems to be no reason to modify the spec in-place, whether on POST or when preparing pod environment. Thoughts?